Implement Getting Tours On Home Screen

Koshake opened the pull request:
Getting and showing tours on home screen.
Fragment, viewModel, adapter implemented

Commit:
feature: HomeViewModel added
feature: Home Fragment added
feature: Home Navigator added

Commit:
feature: Text views for no tours are loaded

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

# Conflicts:
#	app/src/main/java/com/demal/koin/modules/ViewModelModules.kt

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

Commit:
feature: binding for adapter for Home fragment added
feature: adapter in Home fragment
feature: progress bars added
feature: drawable for liked tours

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

# Conflicts:
#	app/src/main/java/com/demal/koin/modules/NavigationModules.kt
#	app/src/main/java/com/demal/navigation/Navigator.kt
#	app/src/main/java/com/demal/navigation/Screens.kt

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

# Conflicts:
#	app/src/main/java/com/demal/koin/modules/NavigationModules.kt
#	app/src/main/java/com/demal/koin/modules/ViewModelModules.kt
#	app/src/main/java/com/demal/navigation/Navigator.kt
#	app/src/main/java/com/demal/navigation/Screens.kt
#	feature-home/build.gradle

Commit:
fix: progress bar for Recomended tours
fix: updating liked tours
refactor: adapter
feat: image loader added to adapter
feat: navigation to tours screen

Commit:
Merge remote-tracking branch 'origin/dev' into feature/home

Commit:
feature: loading images
refactor: adding tour to favorites

Commit:
refactor: init viewmodel in onViewCreated

arsensaliev commented on feature-home/src/main/res/layout/fragment_home.xml:
```diff
@@ -1,83 +1,154 @@
         android:layout_height="wrap_content"
-        android:layout_marginStart="@dimen/margin_home_left"
-        android:layout_marginTop="@dimen/margin_home_textview_top"
+        style="@style/ItemMainTextView"
```
Размеры не соответствует как в фигме. У тебя 18sp в Фигме 20sp.

Koshake replied:
fixed

arsensaliev commented on feature-home/src/main/res/layout/fragment_home.xml:
```diff
@@ -1,83 +1,154 @@
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recommendations_recyclerview"
         android:layout_width="wrap_content"
         android:layout_height="0dp"
```
200dp в фигме.

Koshake replied:
fixed

arsensaliev commented on feature-home/src/main/res/layout/fragment_home.xml:
```diff
@@ -1,83 +1,154 @@
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/activities_recyclerview"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
200dp

Koshake replied:
fixed

arsensaliev commented on feature-home/src/main/res/layout/fragment_home.xml:
```diff
@@ -1,83 +1,154 @@
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/popular_tours_recyclerview"
         android:layout_width="wrap_content"
         android:layout_height="0dp"
```
Размеры не соответствует по фигме. RecyclerView не должен быть выше 200dp. Стоит пересмотреть все размеры и указать как в figma.

Koshake replied:
fixed

arsensaliev commented on feature-home/src/main/res/layout/fragment_home.xml:
```diff
@@ -1,83 +1,154 @@
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recommendations_recyclerview"
         android:layout_width="wrap_content"
         android:layout_height="0dp"
```
Размеры не соответствует по фигме. RecyclerView не должен быть выше 200dp. Стоит пересмотреть все размеры и указать как в figma.

Koshake replied:
fixed

arsensaliev reviewed (changes requested):
Размеры не соответствует по фигме. Карточки item не соответствует как в фигме. Также следует добавить затемнение фотографии. Стоит пересмотреть все размеры и указать как в figma. В Item для картинок используется у вас ImageView а надо по хорошому ShapeableImageView. Посмотрите вариант универсального макета item_tour.xml в core.  И да если не добавить чуть затемнение картинки то белый текст плохо будет виден.

Commit:
fix: RV height changed to 200dp
fix: item home ImageView changed to ShapedImageView
fix: item home margins
fix: header style changed

Commit:
feature: NestedScrollView added to fragment home
feature: item home image darkened 30%
fix: padding end removed from RV
